game-frontend/src/service/game/context.ts

```diff
@@ -34,11 +37,30 @@ export function useWebsocket(): ContextState {
   useEffect(() => {
     if (socket) {
       socket.addEventListener('message', event => {
-        const action: Message = JSON.parse(event.data);
-        dispatch(action);
+        const message: Message = JSON.parse(event.data);
+        dispatch(message);
       });
+      // persist role choice
+      socket.addEventListener('message', event => {
+        const message: Message = JSON.parse(event.data);
+        if (message.type === 'roleResponse') {
+          const { newRole } = (message as RoleResponse).payload;
+          setRole(newRole);
```

Comment:
Storing the role twice kind of defeats the purpose of a `single source of truth`.

Reply:
There is actually no need to store it within the other context - it's quite easy to provide the localstorage state (`usePersistRole`) directly for other components to use.